writer.go

```diff
@@ -114,27 +114,42 @@ func recordID(Key sdk.Data) string {
 	return string(key)
 }
 
-func recordPayload(payload sdk.Change) ([]float32, error) {
+type recordPayload struct {
+	Values       []float32    `json:"values"`
+	SparseValues sparseValues `json:"sparse_values"`
+}
+
+func (r recordPayload) PineconeSparseValues() *pinecone.SparseValues {
+	v := &pinecone.SparseValues{r.SparseValues.Indices, r.SparseValues.Values}
+	return v
```

Comment:
It looks like the method might not be needed, as you're simply returning a value.